packages/amplify-util-uibuilder/src/commands/generateComponents.ts

```diff
@@ -13,12 +13,15 @@ import {
   generateAmplifyUiBuilderIndexFile,
   generateUiBuilderForms,
   generateAmplifyUiBuilderUtilFile,
+  isStudioForm,
+  isFormDetachedFromModel,
+  deleteDetachedForms,
 } from './utils';
 
 /**
  * Pulls ui components from Studio backend and generates the code in the user's file system
  */
-export const run = async (context: $TSContext): Promise<void> => {
+export const run = async (context: $TSContext, eventType: 'PostPush' | 'PostPull'): Promise<void> => {
```

Comment:
I think eventType should be its own type and pull 'PostPush' | 'PostPull' out of the parameters. This will allow for more extensibility when adding other event types in the future such as PrePush, PrePull, PostInit, and others.

Reply:
Again, I agree with the sentiment here, but the events are hooks defined by the platform, so I think this is also out of scope. This is a gap in our own API. I'll create a task to track this in our backlog.